src/main/java/com/google/sps/DataServlet.java

```diff
@@ -169,16 +161,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
     truncatedGraph = currDataGraph.getReachableNodes(nodeNameParam, depthNumber);
 
-    // We set the headers in the following 3 scenarios:
+    if (nodeNameParam.equals("") && truncatedGraph.equals(currDataGraph.graph())) {
```

Comment:
sometimes you use nodeNameParam.length() == 0 and other times .equals(""). Maybe just stick to one or the other for consistency?

Reply:
Good point :)